From 30677a94a969edec58e73da0d0aa3de03a3b6f7e Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Thu, 25 May 2017 17:16:50 +0800
Subject: [PATCH 33/70] android: use ld.bfd instead of ld.gold

We build the kernel with android toolchain. But kernel 4.9 needs
ld.bfd to be linked correctly.
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3142e67d03f1bc..3988531da01386 100644
--- a/Makefile
+++ b/Makefile
@@ -379,7 +379,7 @@ KBUILD_HOSTLDLIBS   := $(HOST_LFS_LIBS) $(HOSTLDLIBS)
 
 # Make variables (CC, etc...)
 AS		= $(CROSS_COMPILE)as
-LD		= $(CROSS_COMPILE)ld
+LD		= $(CROSS_COMPILE)ld$(if $(wildcard $(lastword $(CROSS_COMPILE))ld.bfd),.bfd)
 CC		= $(CROSS_COMPILE)gcc
 CPP		= $(CC) -E
 AR		= $(CROSS_COMPILE)ar
-- 
2.17.1

